src/encoded/static/components/filegallery.js

```diff
@@ -3034,14 +3044,19 @@ class FileGalleryRendererComponent extends React.Component {
             let allFiles = datasetFiles.concat(relatedFiles);
             allFiles = this.filterForInclusion(allFiles);
 
-            // set up filesFilteredByAssembly initially with files from drop down
-            const compiledAnalysis = currentTab === 'browser'
-                ? this.getBrowserAnalysis()
-                : compileAnalyses(context, allFiles, 'choose analysis');
             const dropdown = this.analysisSelectorRef.current;
-            const selectedIndex = currentTab === 'browser'
-                ? this.getBrowserSelectedIndex()
-                : dropdown ? dropdown.selectedIndex : null;
+            let selectedIndex = 0;
+            let compiledAnalysis = compileAnalyses(context, allFiles, 'choose analysis');
+
+            if (currentTab === 'browser') {
+                compiledAnalysis = compiledAnalysis.filter((c) => {
+                    const analysisFiles = filterForVisualizableFiles(c.files.map((f) => allFiles.find((aF) => aF['@id'] === f)));
+                    return analysisFiles.length > 0;
+                });
```

Comment:
Seems like an exact copy of `this.getBrowserAnalysis()`. You couldn’t call that here?

Reply:
Corrected.